source/jobs.c

```diff
@@ -574,27 +544,35 @@ static bool isThingnameTopicMatch( const char * topic,
                                    const char * thingName,
                                    const size_t thingNameLength )
 {
-    char expectedTopicBuffer[ TOPIC_BUFFER_SIZE + 1 ] = { 0 };
+    char expectedTopicBuffer[ TOPIC_BUFFER_SIZE + 1 ] = { '\0' };
     bool isMatch = true;
     size_t start = 0U;
 
-    if( ( topic == NULL ) || ( topicLength == 0 ) )
+    if( ( topic == NULL ) || ( topicLength == 0U ) )
     {
         isMatch = false;
     }
-    else if( ( thingName == NULL ) || ( thingNameLength == 0 ) )
+    else if( ( thingName == NULL ) || ( thingNameLength == 0U ) )
     {
         isMatch = false;
     }
+    else
+    {
+        /* Empty MISRA body */
+    }
 
     if( isMatch )
     {
-        writePreamble( expectedTopicBuffer, &start, TOPIC_BUFFER_SIZE, thingName, thingNameLength );
+        writePreamble( expectedTopicBuffer, &start, TOPIC_BUFFER_SIZE, thingName, ( uint16_t ) thingNameLength );
         ( void ) strnAppend( expectedTopicBuffer, &start, TOPIC_BUFFER_SIZE, topicSuffix, topicSuffixLength );
 
         isMatch = ( size_t ) strnlen( expectedTopicBuffer, TOPIC_BUFFER_SIZE ) ==
                   topicLength;
-        isMatch = isMatch && strncmp( expectedTopicBuffer, topic, topicLength ) == 0;
+        isMatch = isMatch && ( strncmp( expectedTopicBuffer, topic, topicLength ) == 0 );
+    }
+    else
+    {
+        /* Empty MISRA body */
```

Comment:
I do not think that only a solitary `if` needs an `else`. I think the rule is that if you have `if-else if...` ladder, then you must have an `else`.

Just an observation, not a blocker.

Reply:
Looking at the rule, you are correct. I will be creating another PR in this library soon to add the OTA Job Parser from https://github.com/FreeRTOS/Labs-Project-ota-example-for-aws-iot-core and can address this there if that works.

Reply:
Sounds good. Although we might not need to "fix-this" as this is also not a violation :)
I was just making an observation :)